auth-api/src/main/java/io/stargate/auth/api/AuthResource.java

```diff
@@ -87,29 +90,83 @@ public Response createToken(Secret body) {
   @Path("/auth")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Response createToken(Credentials body) {
-    if (body == null) {
+  public Response createToken(Credentials credentials) {
+    if (credentials == null) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide a body to the request"))
           .build();
     }
 
-    if (body.getUsername() == null || body.getUsername().equals("")) {
+    if (credentials.getUsername() == null || credentials.getUsername().equals("")) {
```

Comment:
this logic should be extracted to a separate method because is occurring in multiple places.

Reply:
These are different objects. This uses Credentials but down below it's UsernameCredentials